x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -401,6 +402,14 @@ export const defineGetCspStatusRoute = (
               },
             });
           }
+          await setupCdrDataViews(
```

Comment:
> maybe we should consider setting up the data views on the setup of our plugin and not on the call of an API? What is the downside of it? only the fact that the data views will be created even if users won't use our features?

The downside is that we will not create a data view when new space is created.

Reply:
I can even add more emphasis to what you already mentioned: We intend to use the data view in the scoring task. We cannot rely on users accessing the findings page to enable the scoring task to function.

Reply:
> The downside to this approach is that a new data view will not be created when new space is created.

Got it, I'm wondering if there is a way to hook to the new space creation (similar how we hook to new integrations being installed) and use that hook to install our data views every time a new space is created. 

> I can even add more emphasis to what you already mentioned: We intend to use the data view in the scoring task. We cannot rely on users accessing the findings page to enable the scoring task to function.

I didn't think about the scoring task in this context, but we do expect to use data views there, yes, one more point of why we shouldn't add it as a part of the status API.

Reply:
@maxcold thank you for bringing up these concerns. They are all valid.

Since the dataview can be deleted by the user, I'd like us to be able to recover from it at any point. And I like how the security solution solved it, but we should do it from the server side, not the client's.

I agree we should not couple one API call with another. The best way IMO is to do it on every API call in our plugin. I'm not afraid from any performance hit here. 

I had the idea that we could add the initialization of our dataviews to the context creation, this way we will initialize it on every API call we have.

wdyt?

Reply:
discussed in a f2f meeting, we agreed to check and install our data views if needed in a middleware. That would make sure we do that for every API call we make. One case which is not covered by that process is if we use data view without hitting our API first. Most of our features are guarded with some kind of PLG block, but it's not a must, even more so with 3rd party data in the picture. But worst case users won't see some feature unless they visit one of our pages where we do have API calls. One potential feature under such risk is our data in the user/host/alerts flyouts